### PR TITLE
Disable BuildTimeUnittest

### PR DESCRIPTION
**Pull request by sherryzy**
b/298237650
b/316012657

### CODE REVIEW

File: build/timestamp.gni [hunk 1]
```diff
@@ -28,4 +28,7 @@ if (is_official_build) {
 # changes to keep this up to date.  (Bots run gn on each build, and for devs
 # the timestamp being 100% accurate doesn't matter.)
 # See compute_build_timestamp.py for tradeoffs for picking the timestamp.
-build_timestamp = "123"
+build_timestamp = exec_script(compute_build_timestamp,
```
Comment (andrewsavage1) [inline build/timestamp.gni:31]: In order for reproducible builds to work, we don't want to put a timestamp in our build—if the test is solely checking this build_timestamp value, I think we should disable the test

Comment (sherryzy) [reply build/timestamp.gni:31]: That is what the test is trying to test.

> we don't want to put a timestamp in our build

So is this a permanently or temporarily decision? If we just don't want to put a timestamp in our build, I agree we can just disable the test.

Comment (andrewsavage1) [reply build/timestamp.gni:31]: It's a permanent decision, at least for now. Let's just disable the test

File: base/BUILD.gn [hunk 2]
```diff
@@ -3132,7 +3132,8 @@ test("base_unittests") {
     "big_endian_unittest.cc",
     "bit_cast_unittest.cc",
     "bits_unittest.cc",
-    "build_time_unittest.cc",
+    # Cobalt builds don't include build timestamps.
+    # "build_time_unittest.cc",
```
Comment (andrewsavage1) [inline base/BUILD.gn:3136]: We should either have a conditional checking use_cobalt_customizations that removes the entire source file if we don't want any of it or we should add the DISABLED_ gtest macro if we're disabling just certain tests in the file, that way we can have easier ways of tracking disabled tests

File: base/BUILD.gn [hunk 3]
```diff
@@ -3513,6 +3512,13 @@ test("base_unittests") {
     "vlog_unittest.cc",
   ]
 
+  if (use_cobalt_customizations) {
+    sources += [
+      # Cobalt builds don't include build timestamps.
+      # "build_time_unittest.cc",
```
Comment (andrewsavage1) [inline base/BUILD.gn:3518]: Sorry for the confusion, I meant to leave it in the original list of sources and to add a `sources -=` here to be clear about which files we don't compile

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**